src/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/index.test.tsx

```diff
@@ -1,113 +1,107 @@
 import React from 'react';
-import { fireEvent, render, screen } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { SettingsEditor } from '.';
 import { OpenSearchProvider } from '../../OpenSearchQueryContext';
-import { OpenSearchDatasource } from '../../../../datasource';
 import { OpenSearchQuery } from '../../../../types';
+import { setupMockedDataSource } from '__mocks__/OpenSearchDatasource';
 
 describe('Settings Editor', () => {
   describe('Raw Data', () => {
-    it('Should correctly render the settings editor and trigger correct state changes', async () => {
-      const metricId = '1';
-      const initialSize = '500';
-      const initialOrder = 'desc';
+    it('should render with an editable size field', async () => {
       const query: OpenSearchQuery = {
         refId: 'A',
         query: '',
         metrics: [
           {
-            id: metricId,
+            id: '1',
             type: 'raw_data',
             settings: {
-              size: initialSize,
+              size: '5',
               order: 'desc',
               useTimeRange: true,
             },
           },
         ],
         bucketAggs: [],
       };
+      const datasource = setupMockedDataSource();
 
       const onChange = jest.fn();
 
-      const { rerender } = render(
-        <OpenSearchProvider query={query} datasource={{} as OpenSearchDatasource} onChange={onChange}>
-          <SettingsEditor metric={query.metrics![0]} previousMetrics={[]} />
+      render(
+        <OpenSearchProvider query={query} datasource={datasource} onChange={onChange}>
+          <SettingsEditor metric={query.metrics[0]} previousMetrics={[]} />
         </OpenSearchProvider>
       );
 
-      let settingsButtonEl = screen.getByRole('button', {
-        name: /Size: \d+, Order: \w+$/i,
-      });
-
       // The metric row should have a settings button
-      expect(settingsButtonEl).toBeInTheDocument();
-      expect(settingsButtonEl.textContent).toBe(`Size: ${initialSize}, Order: ${initialOrder}`);
+      expect(screen.getByText('Size: 5, Order: desc')).toBeInTheDocument();
```

Comment:
can you use [`screen.debug()`](https://testing-library.com/docs/dom-testing-library/api-debugging/#screendebug) to print out the component tree to get a sense of how big it is?

also can you time how long the difference is between using `getByRole` vs `getByText`?